.github/workflows/release-helm-chart.yml

```diff
@@ -1,30 +1,64 @@
-# TODO Once distributed chroma is operational, update this to update the
-# OSS helm chart we'll host. For now, just kick off the job which updates
-# hosted-chroma's version.
-
 name: 📦 Release Helm Chart
 
 on:
   push:
-    branches:
-      - main
     paths:
-      - 'k8s/**'
+      - k8s/distributed-chroma/Chart.yaml
   workflow_dispatch:
 
 jobs:
-  release:
+  detect-version-change:
+    name: Detect if version in Chart.yaml was changed
+    runs-on: ubuntu-latest
+    outputs:
+      version_changed: ${{ steps.detect-version-change.outputs.version_changed }}
+      version: ${{ steps.detect-version-change.outputs.version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - name: Detect if version field in Chart.yaml was changed
+        id: detect-version-change
+        shell: bash
+        run: |
+          current=$(git show HEAD:$file | yq ".version")
+          previous=$(git show HEAD^:$file | yq ".version")
+
+          echo "version=$current" >> $GITHUB_OUTPUT
+
+          if [ "$current" != "$previous" ]; then
+            echo "Version field in $file was changed from $previous to $current"
+            echo "version_changed=true" >> $GITHUB_OUTPUT
+          else
+            echo "Version field in $file was not changed"
+            echo "version_changed=false" >> $GITHUB_OUTPUT
+          fi
+        env:
+          file: k8s/distributed-chroma/Chart.yaml
+
+  publish-helm:
+    name: Publish Helm chart
+    needs: detect-version-change
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      AWS_REGION: us-east-1
+    if: ${{ needs.detect-version-change.outputs.version_changed == 'true' || github.event_name == 'workflow_dispatch' }}
```

Comment:
Just a thinking out loud...so, a manual run of this will always trigger a push to ECR... do we want to make sure that the version gets bumped in that case too? Should the manual trigger take a version and just use it?

Reply:
I am envisioning that the manual trigger is used if the automatic publish step fails for some reason. I think the version in `Chart.yaml`should always be the version that's published.